stm/stm.ml

```diff
@@ -2404,7 +2407,8 @@ let new_doc { doc_type ; ml_load_path; vo_load_path; injections; stm_options } =
   let set_option = let open Goptions in function
       | opt, OptionUnset -> unset_option_value_gen ~locality:OptLocal opt
       | opt, OptionSet None -> set_bool_option_value_gen ~locality:OptLocal opt true
-      | opt, OptionSet (Some v) -> set_option_value ~locality:OptLocal (interp_set_option opt) opt v in
+      | opt, OptionSet (Some v) -> set_option_value ~locality:OptLocal (interp_set_option opt) opt v
+      | opt, OptionAppend v -> set_string_option_value_gen ~locality:OptLocal opt v in
```

Comment:
I don't understand the new code, in particular how this `set_string_option_value_gen` is any different from the `set_option_value` above (given that it is precisely defined in terms of `set_option_value`).

Reply:
oops